docs/hub/spaces-handle-url-parameters.md

```diff
@@ -0,0 +1,27 @@
+# How to handle URL parameters in Spaces
+
+You can use URL query parameters as a data sharing mechanism, for instance to be able to deep-link into an app with a specific state.
+
+On a Space page (`https://huggingface.co/spaces/<user>/<app>`), the actual application page (`https://*.hf.space/`) is embedded in an iframe. The query string and the hash attached to the parent page URL are propagated to the embedded app on initial load, so the embedded app can read these values without special consideration.
+
+In contrast, updating the query string and the hash of the parent page URL from the embedded app is slightly more complex.
+If you want to do this in a Docker or static Space, you need to add the following JS code that sends a message to the parent page that has a `queryString` and/or `hash` key.
+
+```js
+const queryString = "...";
+const hash = "...";
+
+window.parent.postMessage({
+    queryString,
+    hash,
+}, "https://huggingface.co");
+```
+
+This is only for Docker or static Spaces.
+For Streamlit apps, Spaces automatically handle the URL parameters sync.
+Gradio is not supporting URL parameters updates/deep-links.
```

Comment:
```suggestion
For Streamlit apps, Spaces automatically syncs the URL parameters. Gradio apps can read the query parameters from the Spaces page, but do not sync updated URL parameters with the parent page.
```

Reply:
Gradio apps _can_ read the query parameters from the Spaces page, e.g. https://huggingface.co/spaces/radames/gradio-url-params?test=%22dsds%22&value=23231